python/dgl/nn/pytorch/factory.py

```diff
@@ -0,0 +1,103 @@
+"""Modules that transforms between graphs and between graph and tensors."""
+import torch.nn as nn
+from ...transform import nearest_neighbor_graph, segmented_nearest_neighbor_graph
+
+def pairwise_squared_distance(x):
```

Comment:
May I ask why these functions has to be `nn.Modules`, it seems they have no parameters and are just wrappers of `nearest_neighbor_graph` and `segmented_nearest_neighbor_graph`.

Reply:
Anything can be a neural network module.  We also have both `torch.nn.functional.relu` and `torch.nn.ReLU` which is a wrapper of the former.